### PR TITLE
Fix panic when OSDK creating crate with `-` in its name

### LINKED ISSUE #1135: OSDK should support creating crate with `-` in its name
As discovered by #1133, `cargo osdk new --kernel my-first-os` will panic due to `my-first-os` contains `-`.

Since `cargo new my-first-os` is allowed, we should fix the problem to keep osdk consistent with cargo.

### PR DESCRIPTION
**Pull request by Marsman1996**
This should fix #1135 

Since `get_src_path()` is only called during the `cargo osdk new` process, its target is only one here. It looks like we do not need to traverse the `targets` array.

The OSDK supports creating crate with - now.
```
root@tca:~/asterinas# cargo osdk new --kernel my-first-os
args.crate_name: my-first-os
cargo_new_lib: my-first-os
    Creating library `my-first-os` package
      Adding `my-first-os` as member of workspace at `/root/asterinas`
note: see more `Cargo.toml` keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
Current directory: "my-first-os"

root@tca:~# cargo osdk new --kernel my-first-os
args.crate_name: my-first-os
cargo_new_lib: my-first-os
    Creating library `my-first-os` package
note: see more `Cargo.toml` keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
Current directory: "my-first-os"
```

### CODE REVIEW

File: osdk/src/commands/new/mod.rs [hunk 1]
```diff
@@ -163,18 +163,13 @@ fn get_manifest_path<'a>(cargo_metadata: &'a serde_json::Value, crate_name: &str
...
-        }
-
+    if targets.len() == 1 {
+        let target = &targets[0];
         let src_path = target.get("src_path").unwrap();
         return src_path.as_str().unwrap();
     }
 
-    panic!("the crate name does not match with any target");
+    panic!("there is no target generated");
```
Comment (tatetian) [inline osdk/src/commands/new/mod.rs:172]: A slightly more readable form of the code:

```rust
assert!(targets.len() == 1, "there must be one and only one target generated");

let target = &targets[0];
let src_path = target.get("src_path").unwrap();
src_path.as_str().unwrap();
```

No indentation. No (explicit) branches.

Comment (Marsman1996) [reply osdk/src/commands/new/mod.rs]: fixed

**Comment by tatetian** [comment]
@StevenJiang1110 Does the fix look good?

**Comment by StevenJiang1110** [comment]
The fix looks fine for me!

@Marsman1996, could you kindly include a test on the issue you resolved in `osdk/test/cli/mod.rs`? This would be helpful in preventing the reoccurrence of the same problem in the future.

The test may be looked like

```rust
#[test]
fn cli_new_crate_with_hyphen() {
    let output = cargo_osdk(&["new", "--kernel", "my-frist-os"]).output().unwrap();
    assert_success(&output);
    assert_stdout_contains_msg(&output, "Creating library `my-first-os` package");
}
```

**Comment by Marsman1996** [comment]
@StevenJiang1110 I added a test. Since "Creating library `my-first-os` package" is not in the stdout, I check whether the folder exists instead. And delete the folder in the end.

File: osdk/tests/cli/mod.rs [hunk 2]
```diff
@@ -50,3 +52,13 @@ fn cli_clippy_help_message() {
...
 }
+
+#[test]
+fn cli_new_crate_with_hyphen() {
+    let output = cargo_osdk(&["new", "--kernel", "my-frist-os"])
+        .output()
+        .unwrap();
+    assert_success(&output);
+    assert!(fs::metadata("my-frist-os").is_ok());
+    fs::remove_dir_all("my-frist-os");
```
Comment (StevenJiang1110) [inline osdk/tests/cli/mod.rs:63]: typo: in `my-frist-os`, frist -> first

Comment (Marsman1996) [reply osdk/tests/cli/mod.rs]: fixed

**Review by StevenJiang1110 — APPROVED** [review]

**Pull request merged**